src/main/kotlin/data/repository/AuditLogRepositoryImpl.kt

```
@@ -11,7 +11,11 @@ class AuditLogRepositoryImpl(
     private val dataProvider: DataProvider<AuditLogEntity>
 ) : AuditLogRepository {
     override fun addAudit(auditLogEntity: AuditLogEntity) {
-        dataProvider.add(auditLogEntity)
+        try {
+            dataProvider.add(auditLogEntity)
+        }catch (e: Exception) {
```

Here you need to return the error direct not need to create another generic exception you can talk a look at user repository 
some times you need to throw exception if you have special cases and the data provider does Handel it